src/plopp/backends/matplotlib/tiled.py

```diff
@@ -0,0 +1,206 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from __future__ import annotations
+
+from typing import Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import gridspec
+
+from ..protocols import FigureLike
+from .static import get_repr_maker
+from .utils import is_interactive_backend, silent_mpl_figure
+
+
+class Tiled:
+    """
+    A tiled figure.
+    This is based on Matpotlib's GridSpec.
+
+    Parameters
+    ----------
+    nrows:
+        Number of rows.
+    ncols:
+        Number of columns.
+    figsize:
+        Figure size (width, height) in inches.
+    **kwargs:
+        Additional arguments passed to :class:`matplotlib.gridspec.GridSpec`.
+
+    Examples
+    --------
+    Create a tiled figure with two plots stacked vertically:
+
+      >>> tiled = pp.tiled(2, 1)
+      >>> tiled[0] = da1.plot()
+      >>> tiled[1] = da2.plot()
+
+    Create a tiled 2x2 figure:
+
+      >>> tiled = pp.tiled(2, 2)
+      >>> tiled[0, 0] = da1.plot()
+      >>> tiled[0, 1] = da2.plot()
+      >>> tiled[1, 0] = da3.plot()
+      >>> tiled[1, 1] = da4.plot()
+
+    Create a tiled figure with two figures side by side and the first is twice as wide:
+
+      >>> tiled = pp.tiled(1, 3)
+      >>> tiled[0, :2] = da1.plot()
+      >>> tiled[0, 2] = da2.plot()
+
+    """
+
+    def __init__(
+        self,
+        nrows: int,
+        ncols: int,
+        figsize: Optional[Tuple[float, float]] = None,
+        **kwargs,
+    ):
+        self.nrows = nrows
+        self.ncols = ncols
+        with silent_mpl_figure():
+            self.fig = plt.figure(
+                figsize=(min(6.0 * ncols, 15.0), min(4.0 * nrows, 15.0))
+                if figsize is None
+                else figsize,
+            )
+        self.gs = gridspec.GridSpec(nrows, ncols, figure=self.fig, **kwargs)
+        self.axes = []
+        self.views = np.full((nrows, ncols), None)
+        self._history = []
+
+    def __setitem__(
+        self,
+        inds: Union[int, slice, Tuple[int, int], Tuple[slice, slice]],
+        view: FigureLike,
+    ):
+        new_view = view.copy(ax=self.fig.add_subplot(self.gs[inds]))
```

Comment:
How about calling `copy_figure` directly here instead of using `view.copy`?
Then we don't need to have `copy` method in all other backend figure types.